SPEC.md

```diff
@@ -0,0 +1,195 @@
+
+
+#### Container Spec
+
+The `containerSpec` field describes edits to be made to the OCI specification. Currently only three kinds of edits can be made to the OCI specification: `devices`, `mounts` and `hooks`.
+
+The `containerSpec` field is referenced in two places in the specification:
+  * At the device level, where the edits MUST only be made if the matching device is requested by the container runtime user.
+  * At the container level, where the edits MUST be made if any of the of the device defined in the `devices` field are requested.
+
+
+The `containerSpec` field has the following definition:
+  * `devices` (array of objects, OPTIONAL) describes the device nodes that should be mounted:
+    * `hostPath` (string, REQUIRED) path of the device on the host.
+    * `containerPath` (string, REQUIRED) path of the device within the container.
+    * `permissions` (string, OPTIONAL) Cgroups permissions of the device, candidates are one or more of:
+      * r - allows container to read from the specified device.
+      * w - allows container to write to the specified device.
+      * m - allows container to create device files that do not yet exist.
+  * `mounts` (array of objects, OPTIONAL) describes the mounts that should be mounted:
+    * `hostPath` (string, REQUIRED) path of the device on the host.
+    * `containerPath` (string, REQUIRED) path of the device within the container.
+    * `readonly` (boolean, OPTIONAL) If set, the mount is read-only.
+    * `propagation` (string, OPTIONAL) Requested propagation mode, candidates are one of:
+      * private - No mount propagation ("private" in Linux terminology).
+      * hostToContainer - Mounts get propagated from the host to the container ("rslave" in Linux terminology).
+      * bidirectional - Mounts get propagated from the host to the container and from the container to the host ("rshared" in Linux terminology).
+  * `hooks` (array of objects, OPTIONAL) describes the hooks that should be ran:
+    * `hookName` is the name of the hook to invoke, if the runtime is OCI compliant it should be one of {createRuntime, createContainer, startContainer, poststart, poststop}.
+      Runtimes are free to allow custom hooks but it is advised for vendors to create a specific JSON file targeting that runtime
+    * `path` (string, REQUIRED) with similar semantics to IEEE Std 1003.1-2008 execv's path. This specification extends the IEEE standard in that path MUST be absolute.
+    * `args` (array of strings, OPTIONAL) with the same semantics as IEEE Std 1003.1-2008 execv's argv.
+    * `env` (array of strings, OPTIONAL) with the same semantics as IEEE Std 1003.1-2008's environ.
+    * `timeout` (int, OPTIONAL) is the number of seconds before aborting the hook. If set, timeout MUST be greater than zero. If not set container runtime will wait for the hook to return.
+
+## Error Handling
+  * Two or more files with identical `kind` values and identical `containerRuntime` field.
```

Comment:
Do we want to address other errors here ?

- non existent kind
- non existent device
- failure to apply any configuration specified in containerSpec by the runtime  (devices, mounts, hooks)

Reply:
https://github.com/container-orchestrated-devices/container-device-interface/pull/9